src/analyzer_status_reporter.ts

```diff
@@ -4,15 +4,15 @@ import { window, StatusBarItem, Disposable } from "vscode";
 import { Analyzer } from "./analyzer";
 import { ServerStatusNotification } from "./analysis_server_types";
 
-let statusBarItem: StatusBarItem;
-let statusShowing: boolean;
-
 export class AnalyzerStatusReporter extends Disposable {
+	statusBarItem: StatusBarItem;
+	statusShowing: boolean;
+
 	constructor(analyzer: Analyzer) {
-		statusBarItem = window.createStatusBarItem();
-		statusBarItem.text = 'Analyzing…';
+		super(() => this.statusBarItem.dispose());
```

Comment:
Heh; I put this further down thinking the reference wouldn't exist till I'd created it but of course it's not called untill later so that make no sense!